.studio/applications-service

```diff
@@ -77,40 +77,39 @@ function applications_integration() {
   A2_SVC_NAME="applications-service" A2_SVC_PATH="/hab/svc/applications-service" go_test "github.com/chef/automate/components/applications-service/integration_test" -v
 }
 
-document "applications_publish_message" <<DOC
-  Publish a message to the NATS server inside the event-gateway.
+document "applications_publish_raw_message" <<DOC
+  Publish a raw message to the Internal NATS Server (event-service)
 
   Example: Publish a hello world message
   --------------------------------------
-  # applications_publish_message "Hello World!"
+  # applications_publish_raw_message "Hello World!"
 DOC
-function applications_publish_message() {
+function applications_publish_raw_message() {
+  local message
+  message="${1-dummy-ping-message}"
+
+  install_if_missing "$(app_service_origin)/applications-service" applications-publisher
+  applications-publisher --internal-nats --port 10140 --raw-message "$message"
+}
+
+document "applications_publish_raw_message_via_gw" <<DOC
+  Publish a raw message to the External NATS Server (event-gateway)
+
+  Example: Publish a hello world message
+  --------------------------------------
+  # applications_publish_raw_message_via_gw "Hello World!"
+DOC
+function applications_publish_raw_message_via_gw() {
```

Comment:
Hmm, so in https://github.com/chef/automate/pull/168 I'm providing a config option that can disable TLS since TLS+NATS isn't yet working on the rust side. In order for the folks working on that stuff to make progress they'll have to disable TLS here. In the referenced PR, there is a flag to disable TLS when you run `applications-publisher`.... but I'm not sure how we best make that work with this helper. Ideas?

Reply:
@danielsdeleo  This has nothing to do with TLS. This is a helper method that sends a *raw* message from both NATs Servers. Internal and External.

Do you see any problem with this PR and yours? Do you think they collide? 🤔

Reply:
@afiune for one, there's git conflict since we both edited the applications-publisher. Not a big deal, just calling it out.

The bigger issue is that applications-publisher now needs to know whether it should use TLS when connecting to the server. The reason for this is that in dev, we use a self-signed certificate for TLS, so we have to disable certificate verification. When we pass the `&tls.Config{}` in to NATS with the `InsecureSkipVerify` option, this has the effect of making the NATS client _demand_ TLS--it will hang up on the server if the server isn't using TLS. So the way I "solved" for that was to just add a CLI flag to the applications publisher so you can turn off TLS. 

If folks on the habitat team are going to meaningfully use this helper in their testing, they will need it to work with TLS disabled (for now) since that's how they need to configure automate (because we're waiting for a proper implementation of TLS in the rust NATS library).

That means they can't use your helper as-is because there's no way to pass in the `--disable-tls` flag to applications publisher, so the command will always fail when testing against automate with TLS disabled at the event-gateway.

I'm not sure what feels most nice-to-use to you as the fix for that. One way to go could be to use `shift` to pop the first argument to the helper and then pass everything else verbatim to the `applications-helper`, similar to how I made the wrapper for `make` work https://github.com/chef/automate/blob/b3055c198189323f10b4da4c10185b1ef9374a1e/.studio/golang#L271-L275